autoscheduler/scraper/management/commands/scrape_courses.py

```diff
@@ -331,55 +330,64 @@ def save_terms(terms, options):
 
     start = time.time()
     now = timezone.now() # use timezone.now() so Django doesn't complain about naive times
-    with transaction.atomic():
-        # Bulk upsert
-        term_models = [Term(code=term, last_updated=now) for term in terms]
 
+    # Bulk upsert
+    with transaction.atomic():
+        # Determine terms to resave
         if options['year'] or options['recent'] or options['term']:
             queryset = Term.objects.filter(code__in=terms)
         else: # If no options are provided, then we're scraping all terms
             queryset = Term.objects.all()
 
+        # Only save terms that actually have courses so that users don't see empty terms
+        terms_with_courses = set(
+            course['term'] for course in
+            Course.objects.distinct('term').filter(term__in=terms).values('term')
```

Comment:
I haven't actually timed it, but neither of those operations should take more than 10-20ms so it doesn't really matter which way we do it— I did it this way because I thought it was pretty clear how the terms with courses are being determined, and we don't have to pass in the scraped courses

Reply:
We talked about it last week, but make this change to search through the list and this will be good to go